selfdrive/ui/qt/offroad/settings.cc

```diff
@@ -217,37 +222,55 @@ QWidget * device_panel() {
   return widget;
 }
 
-QWidget * developer_panel() {
-  QVBoxLayout *main_layout = new QVBoxLayout;
-  main_layout->setMargin(100);
-
-  Params params = Params();
-  std::string brand = params.read_db_bool("Passive") ? "dashcam" : "openpilot";
-  std::vector<std::pair<std::string, std::string>> labels = {
-    {"Version", brand + " v" + params.get("Version", false)},
-    {"Git Branch", params.get("GitBranch", false)},
-    {"Git Commit", params.get("GitCommit", false).substr(0, 10)},
-    {"Panda Firmware", params.get("PandaFirmwareHex", false)},
-    {"OS Version", Hardware::get_os_version()},
-  };
-
-  for (int i = 0; i < labels.size(); i++) {
-    auto l = labels[i];
-    main_layout->addWidget(labelWidget(QString::fromStdString(l.first), QString::fromStdString(l.second)));
+// DeveloperPanel
 
-    if(i+1 < labels.size()) {
+DeveloperPanel::DeveloperPanel() : QFrame() {
+  QVBoxLayout *main_layout = new QVBoxLayout(this);
+  main_layout->setMargin(100);
+  auto dev_params = getParams();
+  for (int i = 0; i < dev_params.size(); i++) {
+    const auto &[name, value] = dev_params[i];
+    LabelWidget *w = new LabelWidget(QString::fromStdString(name), QString::fromStdString(value));
+    labels[QString::fromStdString(name)] = w;
+    main_layout->addWidget(w);
+
+    if(i < (dev_params.size() - 1)) {
       main_layout->addWidget(horizontal_line());
     }
   }
 
-  QWidget *widget = new QWidget;
-  widget->setLayout(main_layout);
-  widget->setStyleSheet(R"(
+  setStyleSheet(R"(
     QLabel {
       font-size: 50px;
     }
   )");
-  return widget;
+}
+
+void DeveloperPanel::showEvent(QShowEvent *event) {
+  updatePanel();
+}
+
+void DeveloperPanel::updatePanel() {
```

Comment:
rename to update? we'll need update functions for a lot of our widgets, so think it makes sense to call them all update